spikeinterface/toolkit/preprocessing/common_reference.py

```diff
@@ -0,0 +1,153 @@
+
+    def __init__(self, recording, reference='median', groups=None, ref_channels=None,
+        local_radius=(2, 8), verbose=False):
+
+        self._kwargs = dict(recording=recording.to_dict(), reference=reference, groups=groups,
+                ref_channels=ref_channels, local_radius=local_radius)
+        
+        num_chans = recording.get_num_channels()
+        neighbors = None
+        # some checks
+        if reference not in ('median', 'average', 'single', 'local'):
+            raise ValueError("'reference' must be either 'median', 'average', 'single' or 'local'")
+
+        if reference == 'single':
+            assert ref_channels is not None, "With 'single' reference, provide 'ref_channels'"
+            if groups is not None:
+                assert len(ref_channels) == len(_groups), \
+                        "'ref_channel' and 'groups' must have the same length"
+            else:
+                if np.isscalar(ref_channels):
+                    ref_channels = [ref_channels]
+                else:
+                    assert len(ref_channels) == 1, \
+                        "'ref_channel' with no 'groups' can be int or a list of one element"
+                ref_channels = np.asarray(ref_channels)
+                assert ref_channels.dtype.kind == 'i',"'ref_channels' must be int"
+        elif reference == 'local':
+            assert groups is None, "With 'local' CAR, the group option should not be used."
+            closest_inds, dist = get_closest_channels(recording)
+            
+            neighbors = {}
+            for i in range(num_chans):
+                mask = (dist[i, :] > local_radius[0]) & (dist[i, :] < local_radius[1])
+                neighbors[i] = closest_inds[i, mask]
+        
+        # tranforms groups (ids) to groups (indices)
+        if groups is not None:
+            groups = [self.ids_to_indices(g) for g in groups]
+
+        BasePreprocessor.__init__(self, recording, )
+        for parent_segment in recording._recording_segments:
+            rec_segment = CommonReferenceRecordingSegment(parent_segment, 
+                        reference, groups, ref_channels, local_radius, neighbors)
+            self.add_recording_segment(rec_segment)
+
+
+class CommonReferenceRecordingSegment(BasePreprocessorSegment):
+    def __init__(self, parent_recording_segment, reference, groups, ref_channels, local_radius, neighbors):
+        BasePreprocessorSegment.__init__(self, parent_recording_segment)
+        
+        self.reference =reference
+        self.groups = groups
+        self.ref_channels = ref_channels
+        self.local_radius = local_radius
+        self.neighbors = neighbors
+        
+        if self.reference  == 'median':
+            self.reducer = lambda x: np.median(x, axis=1)[:, None]
+        elif self.reference  == 'average' :
+            self.reducer = lambda x: np.average(x, axis=1)[:, None]
+        elif self.reference  == 'single' :
+            # no reducer
```

Comment:
If I understand correctly, here you will add a self.reducer corresponding to the average or the median depending of the discussion on the other PR ? Once I have a feedback of my lab, I can post you the answer here if you want. 

Maybe you could also instead of having a reference option have two options : 
reference = global, local or single 
reducer (or a better word) = average, median 
Then the user can select if he wants the average or the median both for a global or a local CAR. What do you think of it ?

Reply:
It was my plan, but I don't want to break too much.

Reply:
To close this question, the answer from my lab is quite balancing with eventually in favor of the median.

Reply:
OK thanks. I will implement what you suggested (having both with diffrent kwargs) with median as default.

Reply:
Does "operator" kwargs souds good ?
`operator='median'/'average'`
Other ideas ?